addons/l10n_it_edi/models/account_move.py

```diff
@@ -701,21 +728,31 @@ def _l10n_it_edi_search_partner(self, company, vat, codice_fiscale, email):
                 return partner
         return self.env['res.partner']
 
-    def _l10n_it_edi_search_tax_for_import(self, company, percentage, extra_domain=None):
+    def _l10n_it_edi_search_tax_for_import(self, company, percentage, extra_domain=None, l10n_it_exempt_reason=None):
         """ Returns the VAT, Withholding or Pension Fund tax that suits the conditions given
             and matches the percentage found in the XML for the company. """
+
         domain = [
             *self.env['account.tax']._check_company_domain(company),
-            ('amount', '=', percentage),
             ('amount_type', '=', 'percent'),
             ('type_tax_use', '=', 'purchase'),
         ] + (extra_domain or [])
 
-        # As we're importing vendor bills, we're excluding Reverse Charge Taxes
-        # which have a [100.0, 100.0, -100.0] repartition lines factor_percent distribution.
-        # We only allow for taxes that have all positive repartition lines factor_percent distribution.
-        taxes = self.env['account.tax'].search(domain).filtered(
-            lambda tax: all(rep_line.factor_percent >= 0 for rep_line in tax.invoice_repartition_line_ids))
+        # As we're importing vendor bills, they can have either normal VAT taxes or Reverse Charge taxes
+        # Reverse charge taxes have 0% as sales but they must be mapped against normal VAT on a purchase
+        # by mapping the sale tax's l10n_it_exempt_reason to a specific tag on the purchase tax.
+        reversed_tax_tag = self._l10n_it_edi_exempt_reason_tag_mapping().get(l10n_it_exempt_reason, '')
+        if not reversed_tax_tag:
+            # Normal VAT taxes have a known percentage and generally have all positive repartition lines
+            domain += [('amount', '=', percentage)]
+            taxes = self.env['account.tax'].search(domain).filtered(
+                lambda tax: all(rep_line.factor_percent >= 0 for rep_line in tax.invoice_repartition_line_ids))
+        else:
+            # Reverse charge taxes have [100, 100, -100] amounts, as the VAT money
+            # is sent to the State by the buyer but then is collected back as Tax credit.
```

Comment:
Why could not we apply the percentage domain?  If in the XML, it only does the positive repartition line, we can deduce the percentage.

Reply:
```suggestion
            # We suppose it comes in from a sale invoice where the tax will be 0%
            # and it will be converted in to a reverse charge tax (which has a negative repartition line)
```

Reply:
It's almost exactly what it's written some lines above, I'll try to be more clear.